File: 14-longest-common-prefix/longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:

        number_of_strings = len(strs)
        base_string = strs[0]

        for ch_index in range(len(base_string)):
            ch = base_string[ch_index]

            for str_index in range(1, number_of_strings):
                str2 = strs[str_index]

                if ch_index == len(str2) or str2[ch_index] != ch:
                    return base_string[:ch_index]

        return base_string

File: 14-longest-common-prefix/longest-common-prefix.java
class Solution {
    
    private String getCommonPrefix(String str1, String str2) {
    
        StringBuilder sb = new StringBuilder();
        
        for(int i = 0; i < str1.length() && i < str2.length(); i++) {
            
            final var ch = str1.charAt(i);
            
            if (ch != str2.charAt(i)) {
                break;
            }

            sb.append(ch);
        }
        
        return sb.toString();
    }
    
    public String longestCommonPrefix(String[] strs) {
        
        var maxCommonPrefix = strs[0];
        
        for (int i = 1; i < strs.length; i++) {
            
            final var str = strs[i];
            maxCommonPrefix = getCommonPrefix(str, maxCommonPrefix);
            
            if (maxCommonPrefix.length()  == 0) {
                break;
            }
        }
        
        return maxCommonPrefix;
    }
}